src/common/time/src/range.rs

```diff
@@ -12,79 +12,182 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::timestamp::TimeUnit;
 use crate::timestamp_millis::TimestampMillis;
+use crate::Timestamp;
 
 /// A half-open time range.
 ///
 /// The time range contains all timestamp `ts` that `ts >= start` and `ts < end`. It is
-/// empty if `start == end`.
+/// empty if `start >= end`.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
What's the definition of time range's "ordering"? I don't get compile error after removing them, it looks like only those basic set operation (and/or) is required. I propose to remove these two derive to avoid ambiguity.
```suggestion
#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
```

Reply:
> What's the definition of time range's "ordering"? I don't get compile error after removing them, it looks like only those basic set operation (and/or) is required. I propose to remove these two derive to avoid ambiguity.

I agree. It does not make sense to sort the order of two ranges.